packages/react-components/react-popover/src/components/PopoverSurface/usePopoverSurface.ts

```diff
@@ -78,6 +78,7 @@ export const usePopoverSurface_unstable = (
     // only close if the event happened inside the current popover
     // If using a stack of inline popovers, the user should call `stopPropagation` to avoid dismissing the entire stack
     if (e.key === 'Escape' && contentRef.current?.contains(e.target as HTMLDivElement)) {
+      e.stopPropagation();
```

Comment:
```suggestion
      // stop propagation to avoid conflicting with other elements that listen for `Escape`
      // e,g: Dialog, Menu
      e.stopPropagation();
```

Let's add a comment similar to the one provided at `usePopoverTrigger` 

https://github.com/microsoft/fluentui/blob/79c08ce5bbf6387b2b18ba4c3d2d5681e0177d4b/packages/react-components/react-popover/src/components/PopoverTrigger/usePopoverTrigger.ts#L51-L53